### PR TITLE
feat: Replace unscheduleAll reducer with API

### PR DESCRIPTION
**Pull request by Bahnschrift**
Waiting on other backend routes to be implemented before adding backend tests.

### CODE REVIEW

File: frontend/src/pages/TermPlanner/OptionsHeader/OptionsHeader.tsx [hunk 1]
```diff
@@ -106,7 +116,7 @@ const OptionsHeader = ({ plannerRef }: Props) => {
             <Popconfirm
               placement="bottomRight"
               title="Are you sure you want to unplan all your courses?"
-              onConfirm={() => dispatch(unscheduleAll())}
+              onConfirm={handleUnscheduleAll}
```
Comment (Bahnschrift) [inline frontend/src/pages/TermPlanner/OptionsHeader/OptionsHeader.tsx:119]: I'm fairly sure this was the only use of the `unscheduleAll` reducer, so I think it should be fine to remove it from `plannerSlice.ts`?

**Comment by Bahnschrift** [comment]
Actually, it seems like tests for this and unscheduleCourse already exist in `backend/server/tests/test_moving_course.py` which both pass. Maybe it would be fine to merge this and #984 as is?

File: frontend/src/pages/TermPlanner/OptionsHeader/OptionsHeader.tsx (hunk 1 above)
Comment (imagine-hussain) [reply frontend/src/pages/TermPlanner/OptionsHeader/OptionsHeader.tsx:119]: lgtm; this can easily be a one-off call

Comment (imagine-hussain) [reply frontend/src/pages/TermPlanner/OptionsHeader/OptionsHeader.tsx:119]: @leonardo-fan does having it as an anonymous function vs passing the function in have any consequence with the Event object being passed in incidentally later on?

**Review by imagine-hussain — APPROVED** [review]
not much to change here, this looks good to me;
feel free to merge.

for testing, can you just make a list of routes that are blocking you so we know what to prioritise? cheers

File: frontend/src/pages/TermPlanner/OptionsHeader/OptionsHeader.tsx (hunk 1 above)
Comment (leonardo-fan) [reply frontend/src/pages/TermPlanner/OptionsHeader/OptionsHeader.tsx:119]: nope this seems all good, handleUnscheduleAll is an anon function anyway.

**Review by leonardo-fan — APPROVED** [review]
lgtm - re: testing do that in a separate pr when you get up to it

**Review by leonardo-fan — CHANGES_REQUESTED** [review]
lgtm - except one thing with the backend route

File: backend/server/routers/planner.py [hunk 2]
```diff
@@ -314,17 +314,13 @@ def unschedule_all(token: str = DUMMY_TOKEN):
         token (str, optional): The user's authentication token. Defaults to DUMMY_TOKEN.
     """
     user = get_user(token)
```
Comment (leonardo-fan) [inline backend/server/routers/planner.py:316]: need a way to get the token from the request - i suggest sending this token via an authorization bearer token header

Comment (leonardo-fan) [reply backend/server/routers/planner.py:316]: you've done it via params in the fe request - which is valid, albeit there's no way in the backend to get that token through the route. however this is not really done in practice

Comment (leonardo-fan) [reply backend/server/routers/planner.py:316]: will do this as a later change

**Review by leonardo-fan — APPROVED** [review]

**Pull request merged**